File: Backend/src/Application/Veiculos/Common/VeiculoResponse.cs
using Domain.Veiculos;

namespace Application.Veiculos.Common;

public class VeiculoResponse
{
    public Guid Id { get; set; }
    public string Placa { get; set; }
    public string Apelido { get; set; }
    public TipoVeiculoEnum TipoVeiculo { get; set; }
    public int? OdometroAtual { get; set; }

    public VeiculoResponse(Veiculo veiculo)
    {
        Id = veiculo.Id;
        Placa = veiculo.Placa;
        Apelido = veiculo.Apelido;
        TipoVeiculo = veiculo.TipoVeiculo;
        OdometroAtual = veiculo.RegistrosOdometro.FirstOrDefault()?.Odometro;
    }
}

File: Frontend/src/screens/veiculo/common/api/types/VeiculoResponse.ts
import type { TipoVeiculoEnum } from "./TipoVeiculoEnum";

export interface VeiculoResponse {
  id: string;
  placa: string;
  apelido: string;
  tipoVeiculo: TipoVeiculoEnum;
  odometroAtual: number | null;
}
